File: updated_code/code/java/1013-partition-array-into-three-parts-with-equal-sum.java
/**
 * This class contains a method that checks if an array of integers can be partitioned into three non-empty parts with equal sums.
 * 
 */
public class Solution {
    /**
     * This method checks if an array of integers can be partitioned into three non-empty parts with equal sums.
     * 
     * @param arr an array of integers
     * @return true if the array can be partitioned, false otherwise
     */
    public boolean canThreePartsEqualSum(int[] arr) {
        int totalSum = 0;
        for (int i = 0; i < arr.length; i++) {
            totalSum += arr[i];
        }
        if (totalSum % 3 != 0) {
            return false;
        }
        int targetSum = totalSum / 3;
        int count = 0;
        int currentSum = 0;
        for (int i = 0; i < arr.length; i++) {
            currentSum += arr[i];
            if (currentSum == targetSum) {
                count++;
                currentSum = 0;
            }
        }
        return count >= 2;
    }
}

File: updated_code/code/python/1013-partition-array-into-three-parts-with-equal-sum.py
from typing import List

class Solution:
    """
    Given an array of integers `arr`, return `true` if we can partition the array into three non-empty parts with equal sums.
    """

    def canThreePartsEqualSum(self, arr: List[int]) -> bool:
        """
        Return `true` if we can partition the array into three non-empty parts with equal sums.
        """
        total_sum = sum(arr)
        if total_sum % 3 != 0:
            return False
        
        target_sum = total_sum // 3
        count = 0
        current_sum = 0
        
        for num in arr:
            current_sum += num
            if current_sum == target_sum:
                count += 1
                current_sum = 0
                if count == 2:
                    break
        
        return count == 2


# Remove trailing whitespace
__str__ = None
__repr__ = None